olga/analytics/views.py

```diff
@@ -82,55 +189,54 @@ def create_instance_data(self, received_data, secret_token):  # pylint: disable=
-            installation_statistics.update(enthusiast_installation_statistics)
-
-        edx_installation_object, _ = EdxInstallation.objects.get_or_create(
-            platform_url=edx_installation['platform_url'],
-            defaults=edx_installation
-        )
+        if statistics_level == 'enthusiast':
+            self.extend_statistics_to_enthusiast_level(received_data, installation_statistics, edx_installation_object)
 
         InstallationStatistics.objects.create(edx_installation=edx_installation_object, **installation_statistics)
 
+    @staticmethod
+    def is_access_token_authorized(access_token):
+        """
+        Check if access token belongs to any EdxInstallation object.
+        """
+        try:
+            EdxInstallation.objects.get(access_token=access_token)
+            logger.info('edX installation with token %s was successfully authorized', access_token)
+            return True
+        except EdxInstallation.DoesNotExist:
+            return False
+
+    @method_decorator(installation_statistics_forms_checker)
     def post(self, request):
         """
-        Receives information from the edx-platform and processes it.
-
-        If the first time gets information from edx-platform, generates a secret token and
-        sends it back to the edx-platform for further data exchange abilities, otherwise
-        updates data in the DB with the new incoming information from the edx-platform.
+        Receive edX installation statistics and create corresponding data in database.
 
-        Returns HTTP-response with status 201, that means object (instance data) was successfully created.
+        Returns HTTP-response with status 201, that means object (installation data) was successfully created.
+        Returns HTTP-response with status 401, that means edX installation is not authorized via token.
         """
-
         received_data = request.POST
-        platform_url = received_data.get('platform_url')
-        secret_token = received_data.get('secret_token')
+        access_token = received_data.get('access_token')
+
+        if self.is_access_token_authorized(access_token):
 
-        if not secret_token:
-            secret_token = uuid.uuid4().hex
-            edx_url = settings.EDX_PLATFORM_POST_URL_LOCAL if settings.DEBUG else (platform_url + '/acceptor_data/')
-            requests.post(edx_url, data={'secret_token': secret_token})
+            logger.info(
+                'edX installation called %s from %s sent statistics after authorization',
+                received_data.get('platform_name'),
+                received_data.get('platform_url')
+            )
+
+            logger.debug(json.dumps(received_data, sort_keys=True, indent=4))
+
+            self.create_instance_data(received_data, access_token)
+            logger.info('Corresponding data was created in OLGA acceptor database.')
+            return HttpResponse(status=httplib.CREATED)
+
+        logger.info(
```

Comment:
make a helper log function from this code please) (no need to write test for it)